modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -192,6 +202,14 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     // TODO: Implement willStopIn
     SequenceView(seq.id, seq.metadata, st, engineSteps(seq), None)
   }
+
+  private def refreshSequenceList(q: engine.EventQueue): Engine.State => Task[Unit] = {
+    st: Engine.State => odbProxy.queuedSequences().flatMap(seqs => EitherT(Nondeterminism[Task].gatherUnordered{
```

Comment:
This is quite hard to follow, perhaps it could be split for better readability? Could you run this inside a for comprehension?